packages/viewer/src/components/viewer-markup/viewer-markup.tsx

```diff
@@ -0,0 +1,385 @@
+    this.selectedMarkupId = undefined;
+  }
+
+  /**
+   * @ignore
+   */
+  @Listen('pointerdown')
+  protected async handleMarkupPointerDown(event: Event): Promise<void> {
+    if (!this.disabled) {
+      const el = event.target as Element;
+      const markups = await this.getMarkupElements();
+      const markup = markups.find((m) => m === el);
+      if (markup?.id != null && markup?.id !== '') {
+        this.selectedMarkupId = el.id;
+      }
+    }
+  }
+
+  /**
+   * @ignore
+   */
+  protected componentDidLoad(): void {
+    this.updatePropsOnMarkupTool();
+  }
+
+  /**
+   * @ignore
+   */
+  protected render(): h.JSX.IntrinsicElements {
+    return (
+      <Host>
+        <slot />
+      </Host>
+    );
+  }
+
+  private createArrowMarkupElement(): HTMLVertexViewerMarkupArrowElement {
+    if (this.arrowTemplateId != null) {
+      const element = stampTemplateWithId(
+        window.document.body,
+        this.arrowTemplateId,
+        isVertexViewerArrowMarkup,
+        () =>
+          console.warn(
+            `Arrow template with ID ${this.arrowTemplateId} not found. Using default arrow element.`
+          ),
+        () =>
+          console.warn(
+            `Arrow template does not contain a vertex-viewer-markup-arrow. Using default arrow element.`
+          )
+      );
+
+      if (element != null) {
+        return element;
+      }
+    }
+
+    return document.createElement('vertex-viewer-markup-arrow');
+  }
+
+  private createCircleMarkupElement(): HTMLVertexViewerMarkupCircleElement {
+    if (this.circleTemplateId != null) {
+      const element = stampTemplateWithId(
+        window.document.body,
+        this.circleTemplateId,
+        isVertexViewerCircleMarkup,
+        () =>
+          console.warn(
+            `Circle template with ID ${this.circleTemplateId} not found. Using default circle element.`
+          ),
+        () =>
+          console.warn(
+            `Circle template does not contain a vertex-viewer-markup-circle. Using default circle element.`
+          )
+      );
+
+      if (element != null) {
+        return element;
+      }
+    }
+
+    return document.createElement('vertex-viewer-markup-circle');
+  }
+
+  private async updatePropsOnMarkups(): Promise<void> {
+    const markup = await this.getMarkupElements();
+    markup.forEach((m) => this.updatePropsOnMarkup(m));
+  }
+
+  private updatePropsOnMarkup(
+    element: HTMLVertexViewerMarkupArrowElement
+  ): void {
+    element.viewer = this.viewer;
+    element.classList.add('viewer-markup__markup');
+  }
+
+  private updatePropsOnMarkupTool(): void {
+    const tool = this.getMarkupTool();
+    if (tool != null) {
+      tool.disabled = this.disabled;
+      tool.arrowTemplateId = this.arrowTemplateId;
+      tool.circleTemplateId = this.circleTemplateId;
+      tool.tool = this.tool;
+      tool.viewer = this.viewer;
+
+      if (this.viewer != null) {
+        this.viewer.cameraControls = tool.cameraControls;
```

Comment:
Looks like this behavior is already handled by the markup tool.

Reply:
Actually looks like this is required to disable the interactions, but I'm not certain why. I've verify that the markup tool is setting `cameraControls` on the viewer it receives to false, but they remain enabled. Curious if you've seen that behavior at all with some of the other components?

Reply:
I suspect it's because `viewer.cameraControls` doesn't respond to being changed. It's working because of the timing of how components are initialized. This may cause problems if you add then remove the tool.

Reply:
I suspect it's because `viewer.cameraControls` doesn't respond to being changed. It's working because of the timing of how components are initialized. This may cause problems if you add then remove the tool.